src/responder/pam/pamsrv_json.h

```diff
@@ -30,64 +30,126 @@
 
 #include "util/sss_pam_data.h"
 
+#define PASSWORD_PROMPT     "Password"
+#define OAUTH2_INIT_PROMPT  "Log In"
+#define OAUTH2_LINK_PROMPT  "Log in online with another device"
+#define PASSKEY_INIT_PROMPT "Insert security key"
+#define PASSKEY_PIN_PROMPT  "Security key PIN"
+#define PASSKEY_TOUCH_PROMPT "Touch security key"
+#define SC_PIN_PROMPT       "Smartcard PIN"
+
+struct auth_data {
+    struct password_data *pswd;
+    struct oauth2_data *oauth2;
+    struct sc_data *sc;
+    struct passkey_data *passkey;
+};
+
+struct password_data {
+    bool enabled;
+    char *prompt;
+};
+
+struct oauth2_data {
+    bool enabled;
+    char *uri;
+    char *code;
+    char *init_prompt;
+    char *link_prompt;
+};
+
+struct sc_data {
+    bool enabled;
+    char **names;
+    char *init_prompt;
+    char **cert_instructions;
+    char *pin_prompt;
+    char **module_names;
+    char **key_ids;
+    char **labels;
+};
+
+struct passkey_data {
+    bool enabled;
+    char *init_prompt;
+    bool key_connected;
```

Comment:
How is key_connected used?

Reply:
Currently it isn't. I added it to make the integration with https://github.com/SSSD/sssd/pull/7983 easier

Reply:
This can be resolved, thanks.